recommender/engines/autoencoders/training/data_preparation_step.py

```diff
@@ -109,67 +92,66 @@ def create_transformer(name):
-        "resource_organisation": service.resource_organisation.name,
-        "scientific_domains": [
-            scientific_domain.name for scientific_domain in service.scientific_domains
-        ],
-        "platforms": [platform.name for platform in service.platforms],
-        "target_users": [
-            (target_user.name, target_user.description)
-            for target_user in service.target_users
-        ],
-        "access_modes": [
-            (access_mode.name, access_mode.description)
-            for access_mode in service.access_modes
-        ],
-        "access_types": [
-            (access_type.name, access_type.description)
-            for access_type in service.access_types
-        ],
-        "trls": [(trl.name, trl.description) for trl in service.trls],
-        "life_cycle_statuses": [
-            (life_cycle_status.name, life_cycle_status.description)
-            for life_cycle_status in service.life_cycle_statuses
-        ],
-    }
-
-    df = df.append(row_dict, ignore_index=True)
+    """It transforms MongoEngine Service object into Pandas dataframe"""
+    service_df = OrderedDict(
+        {
+            "name": service.name,
+            "description": service.description,
+            "tagline": service.tagline.split(", "),
+            "countries": list(service.countries),
+            "categories": [category.name for category in service.categories],
+            "providers": [provider.name for provider in service.providers],
+            "resource_organisation": service.resource_organisation.name,
+            "scientific_domains": [
+                scientific_domain.name
+                for scientific_domain in service.scientific_domains
+            ],
+            "platforms": [platform.name for platform in service.platforms],
+            "target_users": [
+                (target_user.name, target_user.description)
+                for target_user in service.target_users
+            ],
+            "access_modes": [
+                (access_mode.name, access_mode.description)
+                for access_mode in service.access_modes
+            ],
+            "access_types": [
+                (access_type.name, access_type.description)
+                for access_type in service.access_types
+            ],
+            "trls": [(trl.name, trl.description) for trl in service.trls],
+            "life_cycle_statuses": [
+                (life_cycle_status.name, life_cycle_status.description)
+                for life_cycle_status in service.life_cycle_statuses
+            ],
+        }
+    )
+    service_df = pd.DataFrame([service_df])
+
     if save_df:
-        service.dataframe = pickle.dumps(df)
+        service.dataframe = pickle.dumps(service_df)
         service.save()
 
-    return df
+    return service_df
 
 
 def user_to_df(user, save_df=False):
     """It transform MongoEngine User object into Pandas dataframe"""
+    user_df = OrderedDict(
```

Comment:
From Python >= 3.7 and Pandas >= 0.23.0 (does respect ordering when initializing a DataFrame or Series with a dict), dict guarantees ordering. In that case, there is no point in using OrderedDict. So I am switching OrderedDict to dict.